client/components/LearnerViewDirective.js

```diff
@@ -443,6 +475,21 @@ tie.directive('learnerView', [function() {
           $scope.$apply();
         };
 
+        $scope.changeTheme = function() {
+          if ($scope.theme === "Dark") {
+            $rootScope.bodyClass = "night-mode";
+            $scope.codeMirrorOptions.theme = "material";
+            $scope.tieFeedbackWindowClass = "night-mode";
+            $scope.tieQuestionWindowClass = "night-mode";
+          }
+          if ($scope.theme === "Light") {
+            $rootScope.bodyClass = "";
+            $scope.codeMirrorOptions.theme = "default";
+            $scope.tieFeedbackWindowClass = "tie-feedback-window";
```

Comment:
Here and above, it would be better to set a single boolean $scope.isInDarkMode or perhaps $scope.theme and then use ng-class to handle the rest.